source/jormungandr/tests/kirin_realtime_tests.py

```diff
@@ -2196,6 +2196,370 @@ def test_add_new_trip_present_in_navitia_with_a_shift(self):
+        # > No result before the disruption
+        ba_14T22_journey_query = empty_query.format(
+            f='stop_point:stopC', to='stop_point:stopA', dt='20120614T220000'
+        )
+        response, status = self.query_region(ba_14T22_journey_query, check=False)
+        assert 'journeys' not in response
+
+        ba_15T22_journey_query = empty_query.format(
+            f='stop_point:stopC', to='stop_point:stopA', dt='20120615T220000'
+        )
+        response, status = self.query_region(ba_15T22_journey_query, check=False)
+        assert 'journeys' not in response
+
+        ba_16T22_journey_query = empty_query.format(
+            f='stop_point:stopC', to='stop_point:stopA', dt='20120616T220000'
+        )
+        response, status = self.query_region(ba_16T22_journey_query, check=False)
+        assert 'journeys' not in response
+
+        # A new disruption with a detour on departure station to have a pass midnight in advance
+        self.send_mock(
+            "vjA",
+            "20120615",
+            'modified',
+            [
+                UpdatedStopTime(
+                    "stop_point:stopB",
+                    arrival=tstamp("20120616T080100"),
+                    departure=tstamp("20120616T080100"),
+                    arrival_delay=0,
+                    departure_delay=0,
+                    arrival_skipped=True,
+                    departure_skipped=True,
+                    is_detour=True,
+                    message='deleted for detour',
+                ),
+                UpdatedStopTime(
+                    "stop_point:stopC",
+                    arrival_delay=0,
+                    departure_delay=0,
+                    arrival=tstamp("20120615T223000"),
+                    departure=tstamp("20120615T223000"),
+                    is_added=True,
+                    is_detour=True,
+                    message='pass midnight in advance',
+                ),
+                UpdatedStopTime(
+                    "stop_point:stopA",
+                    arrival=tstamp("20120616T080102"),
+                    departure=tstamp("20120616T080102"),
+                    arrival_delay=0,
+                ),
+            ],
+            disruption_id='stop_time_with_detour',
+            effect='detour',
+        )
+
+        # A new disruption is added
+        disruptions_after = self.query_region(disruption_query)
+        assert initial_nb_disruptions + 1 == len(disruptions_after['disruptions'])
+
+        # Now we have 8 vehicle_journeys
+        pt_response = self.query_region('vehicle_journeys')
+        assert initial_nb_vehicle_journeys + 1 == len(pt_response['vehicle_journeys'])
+
+        # After disruption from B to A at 08:00
+        # > vjA circulates the day before
+        response = self.query_region(ba_15T08_journey_query)
+        assert len(response['journeys']) == 1
+        assert response['journeys'][0]['departure_date_time'] == '20120615T080100'
```

Comment:
Currently it's failing on 20120615: this is the (wrong) one that's disrupted (and of course below on 20120616 it's failing too, as it's not disrupted on the good day).